### ISSUE TITLE
TechDraw: Changes to a model's dimensions break angular dimensions in TechDraw documents

### ISSUE THREAD

**Issue by blank-supportgis** [FreeCAD/FreeCAD#17096]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

I've created two views of a Model on a TechDraw page. The views contain lengths, as well as angular dimensions.
When I now change a dimension on my model and update the TechDraw page, the angular dimensions seem to break, throwing an unknown exception
```
14:48:31  1215.58 <App> Document.cpp(3170): Unknown exception in Trichter_Schematik_2#Dimension008 thrown
14:48:31  1215.58 <App> Document.cpp(3170): Unknown exception in Trichter_Schematik_2#Dimension007 thrown
14:48:31  1215.59 <App> Document.cpp(3170): Unknown exception in Trichter_Schematik_2#Dimension005 thrown
14:48:31  1215.6 <App> Document.cpp(3170): Unknown exception in Trichter_Schematik_2#Dimension004 thrown 
```
Deleting the dimensions in question and recreating them fixes the problem.

**Steps to Reproduce**

1. Open [my example file](https://www.dropbox.com/scl/fi/mvvjrzc08sj9zluee9sac/Trichter_Schematik_2.FCStd?rlkey=ozp34s8u0wtvwmervoyhh6bev&st=1rv8np5m&raw=1). 
2. On the variable set `VarSet`, set `Trichterboden->Breite` (funnel bottom -> width) from 380 to 385. This variable is referenced at several places to increase the width of the whole part.
3. Recompute part `Trichter` (funnel).
4. Go to TechDraw page `Page` and click on `Redraw Page`.
5. Above exceptions should be thrown and all angular dimensions should bear a red exclamation mark. Also, the values in the drawings should not change.

### Full version info

```shell
OS: Ubuntu 20.04.6 LTS (ubuntu:GNOME/ubuntu)
Word size of FreeCAD: 64-bit
Version: 1.0.0RC1.38642 (Git) AppImage
Build type: Release
Branch: (HEAD detached at 1.0rc1)
Hash: 60a251354e203bdb924e2190fb9f5e18b48d7362
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: English/United States (en_US)
Installed mods: 
  * btl 0.9.9 (Disabled)
```

### Subproject(s) affected?

None

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by luzpaz** [comment]
Can you reproduce on RC2 ? (https://github.com/FreeCAD/FreeCAD-Bundle/releases/tag/1.0rc2)

**Comment by blank-supportgis** [comment]
There is an RC2? How could I miss that...

Anyway, the problem still persists, I'm afraid. The difference is that recomputation and and reload (steps 3 and 4) are triggered automatically upon exiting the property text field.

**Comment by WandererFan** [comment]
as a work around, go to the Advanced tab of the TechDraw preferences and turn off autocorrect.   This will allow the dimensions to process, even though the current references do not match the save geometry due to the change in
Trichterboden->Breite.  

![autocorrectPreference](https://github.com/user-attachments/assets/eed4133f-ecfb-434e-a0c8-0d6c2154ae36)

**Comment by WandererFan** [comment]
You might want to look into the geometry errors in Body001.  They are not the cause of this problem(??), but they could cause trouble going forward.

![Trichter_Schematik_checkGeom](https://github.com/user-attachments/assets/3c0e58ce-9ebe-423c-b812-942bcb7a496d)

**Comment by WandererFan** [comment]
The exception messages look like they are caused by a missing check in GeometryMatcher::compareBSplines.

**Comment by blank-supportgis** [comment]
> You might want to look into the geometry errors in Body001. They are not the cause of this problem(??), but they could cause trouble going forward.

That's odd. If I select Trichterkoerper (Body001) and run a geometry check, with and without `Run boolean operation check` ticked, I get nothing. 
Why might these errors occur anyway? I'm not aware of making any "reckless" operations.

**Comment by WandererFan** [comment]
Might be the different OCC version,  7.7.2 for you and 7.8 for me. 

I doubt that the errors are from "reckless" operations on your part.  These geometry errors show up from the most mundane operations.  The errors generally don't affect the 3d view since it is only concerned with the outer shell, but if there are subsequent operations like projection or boolean cuts there can be issues.

**Comment by WandererFan** [comment]
I believe this was fixed by PR #17211.

**Comment by luzpaz** [comment]
This commit should be in weekly and in RC3 (when it's released). Please test weekly to make sure this is fixed https://github.com/FreeCAD/FreeCAD-Bundle/releases/tag/weekly-builds 

If you can reproduce, then please notate this ticket and we'll re-open it.
